Physicc/src/bvh.cpp

```diff
@@ -1,42 +1,143 @@
 #include "tools/Tracy.hpp"
+/**
+ * @file bvh.cpp
+ * @brief Constructs a BVH given a list of RigidBody objects.
+ * @author Divyansh Tiwari (divyanshtiwari237)
+ * @author Tirthankar mazumder (wermos)
+ * @bug No known bugs.
+ */
+
+/* -- Includes -- */
+/*bvh header*/
 
 #include "bvh.hpp"
 
 #include <utility>
+#include <algorithm>
 
 namespace Physicc
 {
-	using Iterator = std::vector<RigidBody>::iterator;
-
 	BVH::BVH(std::vector<RigidBody> rigidBodyList)
 		: 	m_head(nullptr),
 			m_rigidBodyList(std::move(rigidBodyList)) 
 	{
 	}
 
-	inline void BVH::buildTree()
+	BoundingVolume::AABB BVH::computeBV(int start, int end)
 	{
 		ZoneScoped;
 
-		buildTree(m_head, m_rigidBodyList.begin(), m_rigidBodyList.end());
+		BoundingVolume::AABB bv(m_rigidBodyList[start].getAABB());
+
+		for (auto i = start + 1; i != end; i++)
```

Comment:
I also changed this line to say `for (std::size_t i = start + 1; i != end; i++)`.